src/openforms/prefill/contrib/objects_api/tests/test_prefill.py

```diff
@@ -0,0 +1,287 @@
+                    {"variable_key": "age", "target_path": ["age"]},
+                ],
+            },
+        )
+
+        prefill_variables(submission=submission)
+        state = submission.load_submission_value_variables_state()
+
+        self.assertEqual(TimelineLogProxy.objects.count(), 1)
+        logs = TimelineLogProxy.objects.get()
+
+        self.assertEqual(logs.extra_data["log_event"], "prefill_retrieve_empty")
+        self.assertEqual(logs.extra_data["plugin_id"], "objects_api")
+        self.assertIsNone(state.variables["lastName"].value)
+        self.assertIsNone(state.variables["age"].value)
+
+    def test_prefilled_values_are_updated_in_the_object(self):
+        """
+        This tests that a created object in the ObjectsAPI prefills the form variables (components) as
+        expected and then (in the same submission) we make sure that we can update the object.
+        """
+        # We manually create the objects instance as if it was created upfront by some external party
+        with get_objects_client(self.objects_api_group) as client:
+            created_obj = client.create_object(
+                record_data=prepare_data_for_registration(
+                    data={
+                        "name": {"last.name": "My last name"},
+                        "age": 45,
+                    },
+                    objecttype_version=3,
+                ),
+                objecttype_url="http://objecttypes-web:8000/api/v2/objecttypes/8e46e0a5-b1b4-449b-b9e9-fa3cea655f48",
+            )
+
+        submission = SubmissionFactory.from_components(
+            initial_data_reference=created_obj["uuid"],
+            components_list=[
+                {
+                    "type": "textfield",
+                    "key": "age",
+                    "label": "Age",
+                },
+                {
+                    "type": "textfield",
+                    "key": "lastName",
+                    "label": "Last name",
+                },
+            ],
+            form__registration_backend="objects_api",
+            with_report=False,
+        )
+
+        user = SuperUserFactory.create()
+        self.client.force_login(user=user)
+        self._add_submission_to_session(submission)
+
+        FormVariableFactory.create(
+            form=submission.form,
+            prefill_plugin="objects_api",
+            prefill_options={
+                "objects_api_group": self.objects_api_group.pk,
+                "objecttype_uuid": "8e46e0a5-b1b4-449b-b9e9-fa3cea655f48",
+                "objecttype_version": 3,
+                "variables_mapping": [
+                    {"variable_key": "lastName", "target_path": ["name", "last.name"]},
+                    {"variable_key": "age", "target_path": ["age"]},
+                ],
+            },
+        )
+
+        prefill_variables(submission=submission)
+
+        # Update the prefilled data
```

Comment:
I probably misunderstood you when you asked a question about registration and that resulted in this part of the test being added. Let's not test that here, sorry for the confusion!